Add within_hours and within_days query options

ebanner opened the pull request:
Addresses https://github.com/TampaDevs/events.api.tampa.dev/pull/57#issuecomment-2254615616

This PR adds `within_hours` and `within_days` query options to the API, so that clients may request events that are `X` hours out or `X` days out from now.

## Notes

I was tempted to abstract the filtering logic into a partial function, but opted to duplicate the code for simplicity.

Also, the browser was requesting a favicon, which was messing with development. I put a check in at the top of the request loop to return, but I'm not sure if it's actually being handled property or not.

manuelosorio reviewed (dismissed):
It seems to be working pretty well. This is a nice solution for days and hours filter.

chtzvt commented:
Hey @ebanner!

I’ve been meaning to get to reviewing your PR and have some feedback. This week is packed for me, but I’ll do my best to knock the review out within the next day or so.

iBotPeaches reviewed (commented):
Taking a look at this @ebanner

I think you probably want these changes in the `filterEventData` from the Util library class. It handles the params and filtering the data down, which these `within_hours=x` and `within_days=x` match.

This appears to work outside of that filtering the data down further, which seems out of place given the param filtering is inside that util.

ebanner commented:
Ah, good call @iBotPeaches - I guess I missed that function. I moved it over.

I moved it over. I think I'd like to do a little bit of testing before merging. I'll make it into a draft for now

chtzvt commented:
@ebanner Great work on this-- stoked to see it merged!

Shoutout to @iBotPeaches also for taking point on code review ❤️
